pkg/operator2/idp.go

```diff
@@ -32,6 +33,11 @@ var (
 	scheme  = runtime.NewScheme()
 	codecs  = serializer.NewCodecFactory(scheme)
 	encoder = codecs.LegacyCodec(osinv1.GroupVersion) // TODO I think there is a better way to do this
+
+	// save latest OIDC password grant check not to bomb the provider with
+	// login requests each sync loop
+	// map is checkedSecretResourceVersion -> passwordGrantsAllowed
+	oidcPasswordChecks = map[string]bool{}
```

Comment:
The worst case scenario is n-times accessing the remote OIDC server in a short interval, where n is the number of workers and I'd expect (n << 10) and the map being written the same value n-times.

I do not see how anyone could trivially add more than one worker for the purpose of what this is doing :thinking: I can add a comment to the map that anyone using it needs to be careful that accessing it is not thread-safe, I'd expect anyone trying to do that would need to tear it off the operator first anyway, so it might be a better spot to comment here?